### PR TITLE
CMP doc: backport improvements on MAC-based protection and critical server authentication

### PR DESCRIPTION
**Pull request by DDvO**
This backports to 3.0 and 3.1 those parts of the doc improvements contained in #21129 that apply also to these stable branches:

* CMP app and API doc: add note on critical server auth on receiving trust anchor certs
* CMP doc: various text improvements, mostly on PBM vs. MAC-based protection

##### Checklist

- [x] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

**Review by paulidale — APPROVED** [review]

**Comment by DDvO** [comment]
@openssl/committers this PR has been waiting for quite some time for the 2nd review, which should be very straightforward

**Comment by DDvO** [comment]
@openssl/committers please let's get this closed soon - the 2nd review should easy,
in particular since the PR for `master` has already been fully approved and merged.

File: doc/man1/openssl-cmp.pod.in
Comment (mattcaswell) [inline doc/man1/openssl-cmp.pod.in:1178]: The "with" here seems superfluous.

**Review by mattcaswell — APPROVED** [review]
Approved with or without the comment above being addressed.

File: doc/man1/openssl-cmp.pod.in
Comment (DDvO) [reply doc/man1/openssl-cmp.pod.in:1178]: Indeed the "M" in "PBM" overlaps with "MAC-based", but "PB" (password-based) does not -
at the latest with the advent of KEM-based message protection, other variants of MAC-based protection are possible, and we already plan to provide KEM-based protection for supporting PQC.

**Comment by DDvO** [comment]
> Approved with or without the comment above being addressed.

Thanks @mattcaswell for our approval -
as explained above, I believe it's better to keep "with PBM".

File: doc/man1/openssl-cmp.pod.in
Comment (mattcaswell) [reply doc/man1/openssl-cmp.pod.in:1178]: I actually meant the first "with", i.e. "using MAC-based protection with PBM or ..." sounds better to me

Comment (DDvO) [reply doc/man1/openssl-cmp.pod.in]: Oh, right, this was a clear mistake (though not part of the fixes done here, which is why I misunderstood your remark).
Fixed.

**Review by mattcaswell — APPROVED** [review]

**Review by paulidale — APPROVED** [review]

**Comment by DDvO** [comment]
Merged to 3.0 and 3.1 - thanks  @paulidale and @mattcaswell

**Pull request closed without merging**
